### PR TITLE
chore: remap arazzo schema to our internal type

### PR DESCRIPTION
**Pull request by DmitryAnansky**
## What/Why/How?

- Remap Arazzo JSON schema to our internal type 
- Add Spot rule sample
- Change makeConfig test util

## Check yourself

- [ ] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by DmitryAnansky** [comment]
@RomanHotsiy / @tatomyr 
There is an open question if we need to keep `export const arazzoSchema` in cli or move it back to Spot?

Keeping it in cli will give us single place to manage schemas and not switching between projects.
Also there is some ongoing activities on Arazzo side - they are going to have official schema and we might use it as source of truth in future.

**Comment by DmitryAnansky** [comment]
> @RomanHotsiy / @tatomyr There is an open question if we need to keep `export const arazzoSchema` in cli or move it back to Spot?
> 
> Keeping it in cli will give us single place to manage schemas and not switching between projects. Also there is some ongoing activities on Arazzo side - they are going to have official schema and we might use it as the source of truth in future.

Discussed offline - I will move it back to Spot codebase.

File: packages/core/__tests__/utils.ts [hunk 1]
```diff
@@ -69,17 +69,24 @@ export function makeConfigForRuleset(
   });
 }
 
-export async function makeConfig(
-  rules: Record<string, RuleConfig>,
-  decorators?: Record<string, DecoratorConfig>,
-  configPath?: string
-) {
+export async function makeConfig({
```
Comment (DmitryAnansky) [inline packages/core/__tests__/utils.ts:72]: @tatomyr I changed this helper function arguments to make it flexible and not pass params in special order and also to handle Arazzo rules case - as it does not merge `rules` with `ArazzoRules` as other specs.

File: packages/core/src/visitors.ts [hunk 2]
```diff
@@ -220,6 +237,23 @@ type Async3FlatVisitor = {
...
+  ExpectSchema?: VisitFunctionOrObject<ExpectSchema>;
+  Replacement?: VisitFunctionOrObject<Replacement>;
+  RequestBody?: VisitFunctionOrObject<RequestBody>;
+  CriteriaObject?: VisitFunctionOrObject<CriteriaObject>;
+  OnSuccessObject?: VisitFunctionOrObject<OnSuccessObject>;
+  OnFailureObject?: VisitFunctionOrObject<OnFailureObject>;
+  Step?: VisitFunctionOrObject<Step>;
+  Steps?: VisitFunctionOrObject<Record<string, Step>>;
+  Workflow?: VisitFunctionOrObject<Workflow>;
+  Workflows?: VisitFunctionOrObject<Record<string, Workflow>>;
```
Comment (tatomyr) [inline packages/core/src/visitors.ts:256]: Isn't it an array instead?

```suggestion
  Workflows?: VisitFunctionOrObject<Workflow[]>;
```

File: packages/core/src/types/arazzo.ts [hunk 3]
```diff
@@ -1,391 +1,308 @@
...
+  required: ['arazzo', 'info', 'sourceDescriptions', 'workflows'],
+  extensionsPrefix: 'x-',
+};
+const Components: NodeType = {
+  properties: {
+    inputs: 'NamedInputs',
+    parameters: {
+      type: 'object',
+      additionalProperties: 'ParameterObject',
+    },
```
Comment (tatomyr) [inline packages/core/src/types/arazzo.ts:23]: Let's have separate names for components (and others too):

```suggestion
    parameters: 'NamedParameters',
    },
```

File: packages/core/src/types/arazzo.ts [hunk 4]
```diff
@@ -1,391 +1,308 @@
...
-  type: 'array',
-  items: parameter,
-} as const;
-export const infoObject = {
-  type: 'object',
+  },
+  required: ['reference'],
+  extensionsPrefix: 'x-',
+};
+const ParameterObject: NodeType = {
```
Comment (tatomyr) [inline packages/core/src/types/arazzo.ts:96]: Why not just this?

```suggestion
const Parameter: NodeType = {
```

File: packages/core/src/types/arazzo.ts [hunk 5]
```diff
@@ -1,391 +1,308 @@
...
-export const criteriaObject = {
-  type: 'object',
+  extensionsPrefix: 'x-',
+};
+const Replacement: NodeType = {
+  properties: {
+    target: { type: 'string' },
+    value: {},
+  },
+  required: ['target', 'replacement'],
```
Comment (tatomyr) [inline packages/core/src/types/arazzo.ts:175]: What is the `replacement` property? Did you mean `value`?

File: packages/core/src/types/oas3.ts [hunk 6]
```diff
@@ -44,7 +44,7 @@ const TagGroup: NodeType = {
   extensionsPrefix: 'x-',
 };
 
-const ExternalDocs: NodeType = {
+export const ExternalDocs: NodeType = {
```
[the remark this answers is not in the document]
Comment (tatomyr) [reply packages/core/src/types/oas3.ts:47]: You mean the schema types? I think it makes sense to import some parts from Oas3 into Oas3.1 as it's the successor. You can try declaring something like `Oas3Schema` and extend it into `Oas3_1Schema`, and then reuse the latter for Arazzo.

**Review by tatomyr — COMMENTED** [review]
Left a few comments.

**Review by JLekawa — APPROVED** [review]

File: packages/core/src/types/oas3.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (DmitryAnansky) [reply packages/core/src/types/oas3.ts:47]: Lets handle this in a separate PR

File: packages/core/src/rules/spot/parameters-no-body-inside-in.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import type { ArazzoRule } from '../../visitors';
+import type { UserContext } from '../../walk';
+
+export const ParametersNoBodyInsideIn: ArazzoRule = () => {
+  return {
+    Parameter: {
+      enter(parameter, { report, location }: UserContext) {
+        if (parameter.in === 'body') {
+          report({
+            message: 'Parameters `in` property not allowed to have `body` value.',
```
Comment (tatomyr) [inline packages/core/src/rules/spot/parameters-no-body-inside-in.ts:10]: Technically it is allowed, but we don't support that. What about this:

```suggestion
            message: 'The `body` value of the `in` property is not supported.',
```

**Review by tatomyr — COMMENTED** [review]
Do we have any docs on Arazzo linting?

**Review by tatomyr — APPROVED** [review]

File: packages/core/src/rules/spot/parameters-no-body-inside-in.ts (hunk 7 above)
Comment (DmitryAnansky) [reply packages/core/src/rules/spot/parameters-no-body-inside-in.ts:10]: 'The `body` value of the `in` property is not supported by Spot'

**Pull request merged**